jolt-core/src/jolt/instruction/sh.rs

```diff
@@ -31,13 +30,20 @@ impl JoltInstruction for SHInstruction {
         C: usize,
         M: usize,
     ) -> Vec<(Box<dyn LassoSubtable<F>>, SubtableIndices)> {
-        // This assertion ensures that we only need two TruncateOverflowSubtables
         // TODO(moodlezoup): make this work with different M
         assert!(M == 1 << 16);
-        vec![(
-            Box::new(IdentitySubtable::<F>::new()),
-            SubtableIndices::from(C - 1),
-        )]
+        vec![
+            (
+                Box::new(IdentitySubtable::<F>::new()),
+                SubtableIndices::from(C - 1),
+            ),
+            (
+                // Not used for lookup, but this implicitly range-checks
```

Comment:
Let's make a utility function for this so it's self documenting and use across all the memory ops that need range-checking.

`fn RangeCheck(C) -> (Box<Subtable<...>>, SubtableIndices)`